src/solver/simulation/sim_calcul_economique.cpp

```diff
@@ -60,15 +60,19 @@ static void importShortTermStorages(
             toInsert.penalizeVariationInjection = st.properties.penalizeVariationInjection;
             toInsert.penalizeVariationWithdrawal = st.properties.penalizeVariationWithdrawal;
             toInsert.name = st.properties.name;
-            toInsert.additional_constraints = st.additional_constraints;
-            for (auto& additional_constraints: toInsert.additional_constraints)
+            for (auto& constr: st.additional_constraints)
             {
-                for (auto& [_, globalIndex,__]: additional_constraints.constraints)
+                if (constr.enabled)
                 {
-                    globalIndex = clusterCumulativeConstraintGlobalIndex;
-                    ++clusterCumulativeConstraintGlobalIndex;
+                    for (auto c: constr.constraints)
```

Comment:
no! catch the constraint by reference

Reply:
you are totally right, without the reference we modify a copy which is not what we want.